service/history/historyEngine.go

```diff
@@ -2013,16 +2018,53 @@ func (e *historyEngineImpl) DeleteWorkflowExecution(
 	if err != nil {
 		return err
 	}
-	defer func() { workflowContext.GetReleaseFn()(retError) }()
+	defer func() { weCtx.GetReleaseFn()(retError) }()
+
+	// Open and Close workflow executions are deleted differently.
+	// Open workflow execution is deleted by terminating with special flag `deleteAfterTerminate` set to true.
+	// This flag will be carried over with CloseExecutionTask and workflow will be deleted as the last step while processing the task.
+
+	// Close workflow execution is deleted using DeleteExecutionTask.
+
+	// DeleteWorkflowExecution is not replicated automatically. Workflow execution must be deleted separately in each cluster.
+	// Although running workflows in active cluster are terminated first and the termination is replicated.
+	// In passive cluster, workflow executions are just deleted in regardless of its status.
 
+	if weCtx.GetMutableState().IsWorkflowExecutionRunning() &&
```

Comment:
With the ability to delete open workflow, the DeleteExecutionsActivity (from namespace delete workflow) should be updated to skip TerminateWorkflow step. 
It can be in follow up PR, just a reminder.

Reply:
Yes, I will do it in separate PR.